llvm/lib/Target/DirectX/DXContainerGlobals.cpp

```diff
@@ -104,6 +112,39 @@ GlobalVariable *DXContainerGlobals::computeShaderHash(Module &M) {
   return GV;
 }
 
+GlobalVariable *DXContainerGlobals::buildContainerGlobal(
+    Module &M, Constant *Content, StringRef Name, StringRef SectionName) {
+  auto *GV = new llvm::GlobalVariable(
+      M, Content->getType(), true, GlobalValue::PrivateLinkage, Content, Name);
+  GV->setSection(SectionName);
+  GV->setAlignment(Align(4));
+  return GV;
+}
+
+GlobalVariable *DXContainerGlobals::buildSingature(Module &M, Signature &Sig,
+                                                   StringRef Name,
+                                                   StringRef SectionName) {
+  std::string Data;
+  raw_string_ostream OS(Data);
```

Comment:
Probably better to use a `raw_svector_ostream` and a `SmallString<256>` as the backing store.

Reply:
Updated.